### DISCUSSION TITLE
How to connect to WSL?

### DISCUSSION THREAD

**Discussion by patelis** [posit-dev/positron#3684] [Q&A] [answered]
I initially downloaded the Ubuntu version of Positron to install on my WSL instance, but when I launched it I was prompted to instead install the Windows version and then proceed to launch Positron through my WSL terminal.

The problem I'm facing is that this does not seem to pick up the R or Python versions I have installed on my Ubuntu WSL distro and the Start Interpreter dropdowns or selectors do not seem to work.

**Comment by wetlandscapes** [comment]
You're doing better than me. I can't even connect to WSL via the Open Remote extensions. ~~I also can't install the Docker extension, and a manual install of the VSIX file just terminates without any messages.~~

Edit: I was able to install that Docker extension after a couple of reboots. I can connect to WSL via the terminal, but like patelis, I don't know how to tell Positron where R or python is installed (it defers to the Windows versions). Perhaps this would not be an issue if using the Open Remote extension? Currently, though, the OSS version of Open Remote doesn't seem to support Positron.

**Reply by patelis** [reply]
> Edit: I was able to install that Docker extension after a couple of reboots. I can connect to WSL via the terminal, but like patelis, I don't know how to tell Positron where R or python is installed (it defers to the Windows versions). Perhaps this would not be an issue if using the Open Remote extension? Currently, though, the OSS version of Open Remote doesn't seem to support Positron.

Yeah, while we can open positron at our WSL folders, it's not really connecting to my Ubuntu VM (assumes it's working on a Windows environment).

On VSCode I use the respective Remote WSL extension that works like a charm, but as noted also by you and @khulenikhil, that's resulting in an error when we try to use the Open Remote version. 
 
I assume this issue might be tackled by this:
https://github.com/posit-dev/positron/issues/2307

If limitation for above issue (for Positron's version of Open Remote SSH) are tackled, I guess maybe they can implement the extension for WSL as well? (Not too sure how much extra effort or an ask this is).

For now I suggest instead of installing the windows version, to install the respective Linux one directly on WSL, that worked for me to identify all python and R installations.

@jmcphers, any clue if by solving remote SSH it would be easy to provide built-in WSL support as well? See error in comment above, I think it's related to your comments on #2307 from March.

**Reply by jmcphers** [reply, marked as answer]
WSL support won't appear automatically with remote SSH support, but they're very similar and once we get remote SSH support we'll have most of the pieces for WSL.

**Reply by tagtag** [reply]
If I am correct, although we can run windows Positron from wsl, Positron does not recognize   R/Python  installed in wsl. Thus, running windows Positron in wsl is  meaningless.  It is better to say explicitly that Positron does not support wsl. Otherwise, people will waste their time.

**Comment by khulenikhil** [comment]
I was able to install the Docker extension and it's working fine.
I tried opening Positron inside WSL using this extension: https://open-vsx.org/extension/jeanp413/open-remote-wsl

But, its startup script is failing. I got the following error:
![image](https://github.com/posit-dev/positron/assets/69512545/f05d30a4-c0de-4fcb-a0bc-5c43caf2bd91)

**Reply by jmcphers** [reply]
Note the URL it is downloading the server from -- it's downloading VSCodium, not Positron. See https://github.com/posit-dev/positron/issues/2307 for more info.

**Reply by kv9898** [reply]
> Note the URL it is downloading the server from -- it's downloading VSCodium, not Positron. See #2307 for more info.

I changed the URL in the `open-remote-wsl` extension in my [fork](https://github.com/kv9898/open-remote-wsl) and published it on Open VSX Registry so it can be downloaded: https://open-vsx.org/extension/kv9898/open-remote-wsl

![Open Remote WSL](https://github.com/user-attachments/assets/30dbc136-3ca2-451f-9a3d-58d7c68b5fe6)
